src/main/java/com/sproutsocial/nsq/Connection.java

```diff
@@ -56,8 +56,8 @@ public Connection(Client client, HostAndPort host) {
     public synchronized void connect(Config config) throws IOException {
         addClientConfig(config);
         Socket sock = new Socket();
-        sock.setSoTimeout(30000);
-        sock.connect(new InetSocketAddress(host.getHost(), host.getPort()), 30000);
+        sock.setSoTimeout(client.getSocketReadTimeoutMillis());
```

Comment:
I'm not sure SO_TIMEOUT should be configurable, a few lines down it is set to heartbeatInterval + 5 secs.
It is only set here because we have to connect to the server first to find out what heartbeatInterval is.